assets/js/data/cart/push-changes.ts

```diff
@@ -65,39 +63,91 @@ const isAddressDirty = <
 	);
 };
 
+type BaseAddressKey = keyof CartBillingAddress | keyof CartShippingAddress;
+
+const getDirtyKeys = < T extends CartBillingAddress | CartShippingAddress >(
```

Comment:
This is the new logic to selectively only get the keys that have changed. Prior to this, the full address object would be pushed, even if a key had no changes, or the address was incomplete.